synapse/handlers/room.py

```diff
@@ -1479,25 +1480,41 @@ async def get_event_for_timestamp(
                         remote_response,
                     )
 
-                    # TODO: Do we want to persist this as an extremity?
-                    # TODO: I think ideally, we would try to backfill from
-                    # this event and run this whole
-                    # `get_event_for_timestamp` function again to make sure
-                    # they didn't give us an event from their gappy history.
                     remote_event_id = remote_response.event_id
-                    origin_server_ts = remote_response.origin_server_ts
+                    remote_origin_server_ts = remote_response.origin_server_ts
```

Comment:
Since we're backfilling the event, we don't have to rely on the unverifiable `origin_server_ts` that the server tells us anymore. 

TODO: Use the `origin_server_ts` from the PDU itself

Reply:
Updated ✅ https://github.com/matrix-org/synapse/pull/13205/commits/b2be2bcdc0ab3d99c7cce91da90eda347f1abd97